src/pipecat/services/background_noise.py

```diff
@@ -0,0 +1,120 @@
+import asyncio
+import json
+import time
+from asyncio import sleep
+from io import BytesIO
+
+import loguru
+
+from pipecat.processors.frame_processor import FrameProcessor, FrameDirection
+from pydub import AudioSegment
+from pipecat.frames.frames import AudioRawFrame, OutputAudioRawFrame, Frame, BotStartedSpeakingFrame, \
+    BotStoppedSpeakingFrame, EndFrame
+
+
+class BackgroundNoiseEffect(FrameProcessor):
+    def __init__(self, websocket_client, stream_sid, music_path):
+        super().__init__(sync=False)
+        self._speaking = True
+        self._audio_task = self.get_event_loop().create_task(self._audio_task_handler())
+        self._audio_queue = asyncio.Queue()
+        self._stop = False
+        self.stream_sid = stream_sid
+        self.websocket_client = websocket_client
+        self.music_path = music_path
+        self.get_music_part_gen = self._get_music_part()
+        self.emptied = False
+
+    async def process_frame(self, frame: Frame, direction: FrameDirection):
+        await super().process_frame(frame, direction)
+
+        if isinstance(frame, BotStartedSpeakingFrame):
+            self._speaking = True
+
+        if isinstance(frame, BotStoppedSpeakingFrame):
+            self._speaking = False
+            self.emptied = False
+
+        if isinstance(frame, AudioRawFrame) and self._speaking:
+            if not self.emptied:
+                self.emptied = True
+                buffer_clear_message = {"event": "clear", "streamSid": self.stream_sid}
+                await self.websocket_client.send_text(json.dumps(buffer_clear_message))
```

Comment:
here we could use the new processor event handlers. for example:

```
self.call_event_handler("on_empty_audio")
```

then, from the pipeline you would do:

```
transport = FastAPIWebsocketTransport()

background_audio = BackgroundAudio("your_path_to_audio_in_format_pcm16000")

@background_audio.event_handler("on_empty_audio", background_audio):
   transport.clear_something()
```